### ISSUE TITLE
Toaster

### ISSUE THREAD

**Issue by serkandurusoy** [mvpspace/redux-starter-kit#7]
I don't like the way we are using the Toaster. it forces us to use component lifecycle methods and thus component classes.

If we can pull the toaster up into the redux store, it would be better. Take a look at `react-s-alert` source code to see how they achieve "globalizing" the alert (toast)

**Comment by ecerroni** [comment]
Do you think is it wort to completely bypass blueprint's toaster and just use `react-s-alert` instead?

Considering the fact that react-s-alert works just great and adding it would be a breeze.

**Comment by serkandurusoy** [comment]
well, why not :)

**Comment by ecerroni** [comment]
Cool :)

**Comment by mvpspace** [comment]
Mhmm I'm having second thoughts on this one :P

Indeed I've realized that I can chain the calls with Toaster so that I can get rid of this awful part:
```
  toast = undefined

  componentDidMount() {
    this.toast = Toaster.create({
      position: Position.TOP,
    })
  }

...

this.toast.show({ message: error.reason, iconName:'pt-icon-error', intent: Intent.DANGER })

```

And just create and show the Toast on the fly like this:
`Toaster.create().show({ message: 'Logged in!!', iconName:'pt-icon-tick', intent: Intent.SUCCESS })`

That's neat and avoids the use of use component lifecycle methods.

I'm wondering if there are any drawbacks, even though it seems a safe call.

**Comment by ecerroni** [comment]
Mhmm I'm having second thoughts on this one :P

Indeed I've realized that I can chain the calls with Toaster so that I can get rid of this awful part:
```
  toast = undefined

  componentDidMount() {
    this.toast = Toaster.create({
      position: Position.TOP,
    })
  }

...

this.toast.show({ message: error.reason, iconName:'pt-icon-error', intent: Intent.DANGER })

```

And just create and show the Toast on the fly like this:
`Toaster.create().show({ message: 'Logged in!!', iconName:'pt-icon-tick', intent: Intent.SUCCESS })`

That's neat and avoids the use of use component lifecycle methods.

I'm wondering if there are any drawbacks, even though it seems a safe call.

**Comment by serkandurusoy** [comment]
`this.toast....` implies you have a context (this) which implies a class
component

**Comment by ecerroni** [comment]
Yep, this.toast.... has gone. I don't use it anymore.

I'll push the changes so you can check the new code.

**Comment by ecerroni** [comment]
Let's reopen this one if it's still not how it should work :)

**Comment by serkandurusoy** [comment]
examine the elements tab of the developer tools while creating new toasts.
and see if they get properly cleaned up.

furthermore, see what happens

* when multiple toast are generated quickly
* when multiple toasts are generated from different components on the page
(new one generated a) before and b) after the old one disappears)
* when you switch a route when there's a visible toast on the page
* when swtiching to a new route and displaying a toaster while the old
route has created a toaster and it is still visible
* when there are high z-index elements on the page, like a modal for
example. ie, try to show a toaster from within a modal

**Comment by ecerroni** [comment]
Basically they have a standard behavior: multiple toasts stack on each other.

So in all cases you mentioned we see that kind of behavior, e.g. like in Plutio.

However now that I spent a while fighting with it I may override the default behavior and add different ones that are more in line with our needs.

For instance besides its own default params I already added a `routeSwitch: true` new parameter in `Toast.show()` that clears all the previous toast if the user is switching between routes. We can name it whatever works best...

I may also limit the number of multiple toasts shown at the same time, etc.

What kind of behaviors would we like to have for each use cases?
